File: Exercism/ruby/resistor-color-duo/resistor_color_duo.rb
=begin
Write your code for the 'Resistor Color Duo' exercise in this file. Make the tests in
`resistor_color_duo_test.rb` pass.

To get started with TDD, see the `README.md` file in your
`ruby/resistor-color-duo` directory.
=end

class ResistorColorDuo

  def self.value(colours)
    values = ["black", "brown", "red", "orange", "yellow", "green", "blue", "violet", "grey", "white"]
    return colours.map { |colour| values.index(colour) }[0,2].join.to_i
  end
end

File: Exercism/dart/resistor-color-duo/lib/resistor_color_duo.dart
class ResistorColorDuo {
    var colors = [
      "black", "brown", "red", "orange", "yellow",
      "green", "blue", "violet", "grey", "white"
    ];

    int value(List<String> resistor) {
      return int.parse(resistor.map(
          (it) => colors.indexOf(it)
        ).take(2).toList().join());
    }
}
